src/thing.rs

```diff
@@ -808,20 +794,10 @@ impl AvailableAction {
     /// Returns a boolean indicating validation success.
     fn validate_action_input(&self, input: Option<&serde_json::Value>) -> bool {
         let mut scope = json_schema::Scope::new();
-        let validator = if self.metadata.contains_key("input") {
-            let mut schema = self
-                .metadata
-                .get("input")
-                .unwrap()
-                .as_object()
-                .unwrap()
-                .clone();
-            if schema.contains_key("properties") {
-                let properties = schema
-                    .get_mut("properties")
-                    .unwrap()
-                    .as_object_mut()
-                    .unwrap();
+        let validator = if let Some(input) = self.metadata.get("input") {
+            let mut schema = input.as_object().unwrap().clone();
+            if let Some(properties) = schema.get_mut("properties") {
+                let properties = properties.as_object_mut().unwrap();
```

Comment:
```suggestion
        let validator = if let Some(Value::Object(input)) = self.metadata.get("input") {
            let mut schema = input.clone();
            if let Some(Value::Object(properties)) = schema.get_mut("properties") {
```
Same idea as in the suggestion for property.rs

Reply:
I think since this would occur as a result of a server-side implementation error, it should stay a hard error instead of a silent fall-through to no validation.